server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreBaseIntegTestCase.java

```diff
@@ -135,14 +136,17 @@ protected Settings nodeSettings(int nodeOrdinal) {
             segmentRepoPath = randomRepoPath().toAbsolutePath();
             translogRepoPath = randomRepoPath().toAbsolutePath();
         }
+        Settings.Builder settingsBuilder = Settings.builder();
         if (clusterSettingsSuppliedByTest) {
-            return Settings.builder().put(super.nodeSettings(nodeOrdinal)).build();
+            settingsBuilder.put(super.nodeSettings(nodeOrdinal));
         } else {
-            return Settings.builder()
-                .put(super.nodeSettings(nodeOrdinal))
-                .put(remoteStoreClusterSettings(REPOSITORY_NAME, segmentRepoPath, REPOSITORY_2_NAME, translogRepoPath))
-                .build();
+            settingsBuilder.put(super.nodeSettings(nodeOrdinal))
+                .put(remoteStoreClusterSettings(REPOSITORY_NAME, segmentRepoPath, REPOSITORY_2_NAME, translogRepoPath));
         }
+        if (randomBoolean()) {
+            settingsBuilder.put(RemoteStoreSettings.CLUSTER_REMOTE_SEGMENT_SEPARATE_METADATA_SEGMENTINFOS_SETTING.getKey(), true);
```

Comment:
@sachinpkale just confirming this segment infos snapshot is different than segments_N file which gets created at the commit time. this is the one which have latest segment info at the time of refresh?